src/App.tsx

```diff
@@ -537,15 +567,19 @@ function App() {
                         }
                         content={
                             <ContentPanel
-                                title={moduleNames[currentModule]}
                                 {...content[currentModule][page]}
-                                finishButton={
-                                    (canDetermineEquilibrium &&
-                                        page <
-                                            PAGE_NUMBER_3D_EXAMPLE[
-                                                currentModule
-                                            ]) ||
-                                    content[currentModule][page].finishButton
+                                currentModule={currentModule}
+                                nextButton={
+                                    (canDetermineKd &&
+                                        content[currentModule][page].section ===
+                                            Section.Experiment) ||
+                                    content[currentModule][page].nextButton
+                                }
+                                nextButtonText={
+                                    lastPageOfExperiment
+                                        ? "Finish"
```

Comment:
creates a finish button once they've gotten to the end of the experiment (which is a dynamic state, not a page number)